### ISSUE TITLE
Segmentation fault running hadoop image, only on mac and only after update

### ISSUE THREAD

**Issue by dantswain** [docker/for-mac#2492]
### Expected behavior

Expect that running
```
docker run --rm -it harisekhon/hadoop:2.7
```
launches a self-contained hadoop container.

### Actual behavior

The container fails to start, I get segfaults in the logs:
```
localhost: /hadoop-2.7.4/sbin/hadoop-daemon.sh: line 131:   227 Segmentation fault      nohup nice -n $HADOOP_NICENESS $hdfsScript --config $HADOOP_CONF_DIR $command "$@" > "$log" 2>&1 < /dev/null
```

### Information

  - Full output of the diagnostics from "Diagnose & Feedback" in the menu

```
Docker for Mac: version: 17.12.0-ce-mac49 (d1778b704353fa5b79142a2055a2c11c8b48a653)
macOS: version 10.13.2 (build: 17C205)
logs: /tmp/B111284B-1E44-4C4D-80A1-832617C6131E/20180123-142921.tar.gz
[OK]     db.git
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     kubernetes
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     slirp
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[OK]     docker-cli
[OK]     menubar
[OK]     disk
```
I am still able to run this container on linux hosts and I have checked with colleagues that are on previous versions of Docker for Mac for which it still works.  This started failing for me within the last few days (since I last updated).  I have tried resetting my installation.

### Steps to reproduce the behavior

Run `docker run --rm -it harisekhon/hadoop:2.7` and note segfaults in the logs.

**Comment by gsong** [comment]
Possibly related to #2542.

**Comment by djs55** [comment]
@dantswain thanks for your report. We did revert the default back to `qcow2` for a while but have since switched the default back to `raw` in edge (and it will be `raw` in the next stable). The underlying bug appeared to be in APFS itself and was fixed in macOS 10.13.4. We'll only use `qcow2` where there is no APFS (old macOS or no SSD) or where the version is buggy (macOS 10.13.0 ... 10.13.3)
